crates/orchestrator/src/tests/jobs/da_job/mod.rs

```diff
@@ -1,64 +1,310 @@
+                internal_id.to_string(),
+                Uuid::default()
+            )
+            .to_string();
+
+            assert_eq!(e.to_string(), expected);
+        }
+    }
+    state_update_mock.assert();
+    Ok(())
 }
 
+/// Tests successful DA Job processing with valid state update and nonces files.
+/// Mocks DA client to simulate environment and expects job to process without errors.
+/// Validates the successful job processing by checking the return message "Done".
+/// Asserts correct behavior by comparing the received and expected success messages.
 #[rstest]
+#[case(
+    "src/tests/jobs/da_job/test_data/state_update/638353.txt",
+    "src/tests/jobs/da_job/test_data/nonces/638353.txt",
+    "638353"
+)]
 #[tokio::test]
-async fn test_verify_job(#[from(default_job_item)] mut job_item: JobItem) {
+async fn test_da_job_process_job_success(
+    #[case] state_update_file: String,
+    #[case] noances_file: String,
+    #[case] internal_id: String,
+) -> Result<()> {
+    // Mocking DA client calls
     let mut da_client = MockDaClient::new();
-    da_client.expect_verify_inclusion().times(1).returning(|_| Ok(DaVerificationStatus::Verified));
+    da_client.expect_publish_state_diff().with(always(), always()).returning(|_, _| Ok("Done".to_string()));
+    da_client.expect_max_blob_per_txn().with().returning(|| 6);
+    da_client.expect_max_bytes_per_blob().with().returning(|| 131072);
+
+    let server = TestConfigBuilder::new().mock_da_client(Box::new(da_client)).build().await;
+    let config = config().await;
+
+    let state_update = read_state_update_from_file(state_update_file.as_str()).expect("issue while reading");
+
+    let state_update = serde_json::to_value(&state_update).unwrap();
+    let response = json!({ "id": 1,"jsonrpc":"2.0","result": state_update });
+
+    get_nonce_attached(&server, noances_file.as_str());
+
+    let state_update_mock = server.mock(|when, then| {
+        when.path("/").body_contains("starknet_getStateUpdate");
+        then.status(200).body(serde_json::to_vec(&response).unwrap());
+    });
+
+    let response = DaJob
+        .process_job(
+            config.as_ref(),
+            &mut JobItem {
+                id: Uuid::default(),
+                internal_id: internal_id.to_string(),
+                job_type: JobType::DataSubmission,
+                status: JobStatus::Created,
+                external_id: ExternalId::String(internal_id.to_string().into_boxed_str()),
+                metadata: HashMap::default(),
+                version: 0,
+            },
+        )
+        .await;
 
-    let config = init_config(None, None, None, Some(da_client), None, None, None).await;
-    assert!(DaJob.verify_job(&config, &mut job_item).await.is_ok());
+    if let Ok(message) = response {
+        assert_eq!(message, eyre!("Done").to_string());
+    }
+
+    state_update_mock.assert();
+    let _ = drop_database().await;
```

Comment:
later, when we move `drop_database` inside the config builder this part will change

Reply:
Yes, we'll have to do a thorough check for the same
Here's an issue to keep track of the same : #64